bin/pycbc_live

```diff
@@ -309,27 +309,33 @@ class LiveEventManager(object):
 
     def check_singles(self, results, data_reader, psds, f_low):
         active = [k for k in results if results[k] != None]
-        if len(active) == 1:
-            ifo = active[0]
+        for ifo in active:
             single = sngl_estimator[ifo].check(results[ifo], data_reader[ifo])
 
-            if single is not None:
-                fud = self.compute_followup_data([ifo], single, data_reader,
-                                                 bank, [])
-                event = SingleCoincForGraceDB([ifo], single, bank=bank,
-                                              psds=psds, followup_data=fud,
-                                              low_frequency_cutoff=f_low,
-                                              channel_names=args.channel_name)
-
-                end_time = int(single['foreground/%s/end_time' % ifo])
-                fname = 'single-%s-%s.xml.gz' % (ifo, end_time)
-                fname = os.path.join(self.path, fname)
-                logging.info('Single-detector candidate! Saving as %s', fname)
-                if args.enable_single_detector_upload:
-                    event.upload(fname, gracedb_server=args.gracedb_server,
-                                 testing=self.gracedb_testing)
-                else:
-                    event.save(fname)
+            if single is None:
+                continue
+            fud = self.compute_followup_data([ifo], single, data_reader,
+                                             bank, [])
+            event = SingleCoincForGraceDB([ifo], single, bank=bank,
+                                          psds=psds, followup_data=fud,
+                                          low_frequency_cutoff=f_low,
+                                          channel_names=args.channel_name)
+
+            end_time = int(single['foreground/%s/end_time' % ifo])
+            fname = 'single-%s-%s.xml.gz' % (ifo, end_time)
+            fname = os.path.join(self.path, fname)
+            logging.info('Single-detector candidate! Saving as %s',
+                         fname)
+            ifar = single['foreground/ifar']
+            # apply a trails factor of the number of active detectors
+            ifar /= len(active)
+            single['foreground/ifar'] = ifar
```

Comment:
I think the application of the trials factor (lines 329-332) should go before instantiating the `SingleCoincForGraceDB`. Otherwise GraceDB will show the uncorrected FAR, right?

Also trails → trials.

Reply:
Yes that's right, don't know why I put it afterwards. Sorted now